### DISCUSSION TITLE
Event Series Modelling

### DISCUSSION THREAD

**Discussion by saumier** [culturecreates/artsdata-planet-lavitrine#2] [Modelling discussions] [answered]
Event Series
===============
To match the LaVitrine data model, the offers, attendance mode and event status properties are moved from "Spectacle" (schema:EventSeries) to "Représentation" (schema:Event).

The Artsdata data model supports this when the data is available.  However, for Event Series (multiple dates for the same show) a majority of websites do not have the data for each "Représentation".  When the data is not available for each "Représentation", the data pipeline to convert from the Artsdata data model to the LaVitrine data model will assume that each "Représentation" (each date) inherits the offers, attendance mode and event status from the "Spectacle" (schema:EventSeries). 

Background Questions
------------
1. Should the Artsdata output pipeline for LaVitrine make the assumption that if the source website only has offers, attendance mode and event status properties in the Event Series, then it should be copied into each of the subEvents? --> **YES**
2. Should an Event with only a single date have a single sub-event added and have the offers, attendance mode and event status properties moved inside to make the model? Then the top level events could be renamed "Spectacle" and the sub-events "Représentation" like in [Référentiel des métadonnées descriptives du spectacle](https://docs.google.com/document/d/1o1wtOwpQEMOoELC2TiI3eacNYwgbJeL1yAADix4a--0/edit) ? --> **YES**

**Comment by christianroy** [comment]
We just discussed Q1 with @GenFab28 and we agree with "yes", as long as this is scoped to a "source website". Which leads to a clarification question. In a situation where the same "spectacle" has "représentations" on multiple websites, will the output feed create one EventSeries with a subEvent for each "représentations", or one EventSeries for each website, each with a few subEvents? As a more concrete example, if we had the two following shows in a single feed, would they be modeled as two different EventSeries, or a single one?

* https://grandtheatre.qc.ca/programmation/led-zeppelin-symphonic/
* https://placedesarts.com/fr/evenement/led-zeppelin-symphonic

**Reply by saumier** [reply]
They will be modelled as 2 different "Spectacles/EventSeries". GTQ will have a "spectacle" URI-1 and a "représentation" URI-2, and PDA will have a "spectacle" URI-3 and a "représentation" URI-4.  

We can explore linking spectacle URI-1 on GTQ and spectacle URI-3 on PDA using schema:sameAs as an optional step if needed in the future.

**Reply by christianroy** [reply]
@saumier all good!

**Comment by christianroy** [comment]
For Q2, we think the answer should be yes. It makes a fully coherent model and it is closer to how it is stored in the Vitrine database.

**Reply by saumier** [reply]
I have opened issue https://github.com/culturecreates/artsdata-planet-lavitrine/issues/8 and will make this change to the Artsdasta data export pipeline for all websites.

**Comment by saumier** [comment, marked as answer]
To match the LaVitrine data model, the offers, attendance mode and event status properties are moved from "Spectacle" (schema:EventSeries) to "Représentation" (schema:Event).

All Events from Artsdata are converted to EventSeries with one "Représentation".  This is to facilitate the task for developers to consume the JSON. 

See the JSON-LD Frame for the placement of properties https://github.com/culturecreates/artsdata-planet-lavitrine/blob/main/frame/lavitrine_event_frame.jsonld
